bot/exts/moderation/infraction/management.py

```diff
@@ -348,13 +339,17 @@ async def cog_check(self, ctx: Context) -> bool:
         return all(checks)
 
     # This cannot be static (must have a __func__ attribute).
-    async def cog_command_error(self, ctx: Context, error: Exception) -> None:
-        """Send a notification to the invoking context on a Union failure."""
+    async def cog_command_error(self, ctx: Context, error: commands.CommandError) -> None:
+        """Handles errors for commands within this cog."""
         if isinstance(error, commands.BadUnionArgument):
             if discord.User in error.converters:
                 await ctx.send(str(error.errors[0]))
                 error.handled = True
 
+        elif isinstance(error, commands.BadArgument):
+            await ctx.send(f":x: Could not find a corresponding infraction for `{error.args[1]}`.")
```

Comment:
This assumes that BadArgument will only ever be raised from the infraction converter.

Perhaps this should be a custom error such as `InfractionConversionError` which will inherit from dpy's `ConversionError`

Reply:
Yeah, I've made an `InvalidInfraction` error and decided to also slightly improve the error messages (invalid id and 404). Will push these changes shortly.

![example usage](https://user-images.githubusercontent.com/47674925/131746073-c394b9b9-781f-44e9-8641-0dec54e2516b.png)

Reply:
Also going to change "...infraction for `{number}`" to `"...infraction with id `{number}`